src/backend/dx12/src/lib.rs

```diff
@@ -1081,13 +1098,34 @@ impl hal::Instance<Backend> for Instance {
                     max_bound_descriptor_sets: MAX_DESCRIPTOR_SETS as u16,
                     max_descriptor_set_uniform_buffers_dynamic: 8,
                     max_descriptor_set_storage_buffers_dynamic: 4,
-                    max_per_stage_descriptor_sampled_images: d3d12::D3D12_COMMONSHADER_INPUT_RESOURCE_REGISTER_COUNT as _,
-                    max_per_stage_descriptor_samplers: d3d12::D3D12_COMMONSHADER_SAMPLER_REGISTER_COUNT as _,
-                    max_per_stage_descriptor_storage_buffers: 4,
-                    max_per_stage_descriptor_storage_images: 4,
-                    max_per_stage_descriptor_uniform_buffers: d3d12::D3D12_COMMONSHADER_CONSTANT_BUFFER_API_SLOT_COUNT as _,
+                    max_descriptor_set_sampled_images: full_heap_count,
+                    max_descriptor_set_storage_buffers: uav_limit,
+                    max_descriptor_set_storage_images: uav_limit,
+                    max_descriptor_set_uniform_buffers: full_heap_count,
+                    max_descriptor_set_samplers: d3d12::D3D12_MAX_SHADER_VISIBLE_SAMPLER_HEAP_SIZE as _,
+                    max_per_stage_descriptor_sampled_images: match features.ResourceBindingTier {
```

Comment:
why is this not `d3d12::D3D12_COMMONSHADER_INPUT_RESOURCE_REGISTER_COUNT` ?

Reply:
With both this and `max_per_stage_descriptor_samplers`, you have the problem of descriptor arrays. `Texture2D<float> array[10000]` counts 10k against this limit, but only uses one register. As far as I'm aware there is no way to express "max_descriptor_registers" in this limit set as vulkan doesn't seem to have a direct analogue.

Reply:
wow, that's unfortunate! so many limits, yet here there is a big problem now for D3D12 and Metal backends :/